res/huroutes.js

```diff
@@ -635,7 +656,25 @@ var navigation = {
         return this.provs[localStorage.navprovider] ? localStorage.navprovider : Object.keys(this.provs)[0];
     },
     /** Returns a navigation link. */
-    getLink: function(coord) { return this.provs[this.getId()](coord); }
+    getLink: function(coords, reverse) {
+        const dst = reverse ? coords[0] : coords[coords.length - 1];
+        const wpts = this._getWaypoints(coords, reverse);    
+        return this._getProv().getLink(dst, wpts);
+    },
+    /** Returns the currently selected navigation provider. */
+    _getProv: function() { return this.provs[this.getId()]; },
+    /** Returns a set of waypoints as requested by the link provider. */
+    _getWaypoints: function(coords, reverse) {
+        const count = this._getProv().waypoints;
+        if (!count)
+            return [];
+        let idx = reverse ? coords.length - 1 : 0;
+        const step = Math.floor(coords.length / count) * (reverse ? -1 : 1);
+        let wpts = [];
+        for (i = 0; i < count; ++i, idx += step)
+            wpts.push(coords[idx]);

```

Comment:
coords.length = 59
reverse = false
count = 10

step will be 5

`coords[58]` will be used in `getLink` as that is the `dst`

this method will use 0, 5, 10 ... 45 in the for loop

so there will be a bigger gap between 45 and 58 (even with `i <= count` it would be 50 vs 58)

shouldn't it use `Math.round` instead? that way (count - 1) * step won't be that far away (0, 6, 12 ... 42, 48, 54) (I mean `a <= Math.ceil(a / b) * b < a + b` (max diff is b - 1) and ` a - b < Math.floor(a / b) * b <= a` (max diff is b - 1) meanwhile `a - b / 2 < Math.round(a / b) * b <= a + b / 2` (max diff is b / 2)

although this fixes the 59 (54 vs 59) with coords.length = 54, the gap will be still pretty big 45 vs 54

.... so how about only rounding at the very end by using `step = coords.length / count` as is and do `coords[Math.round(idx)]`? it would be, 0, 6, 12, 18, 24, 30, 35, 41, 47, 53, (58)? and it would be 43, 49, 54 for 54 length

Reply:
Yes, I know that this is not precise. But the vertices of the road arent precise either. I think that this should still provide a sufficient precision. The algorithm is faster and shorter when using integers and flooring.

Reply:
"The algorithm is faster and shorter when using integers and flooring."

How many waypoints and "count" are we talking about that it matters?